docs/spec/README.md

```diff
@@ -0,0 +1,52 @@
+# Source Controller Proposal
+
+## Context
+
+The desired state of a cluster is made out of Kubernetes objects, these objects are expressed in `.yaml` format and 
+are applied on the cluster by operators running inside the cluster. An operator's role is to fetch the Kubernetes
+objects, run transformations on them and reconcile the cluster state with the resulting manifest.
+
+For an operator to acquire the resources that make up the desired state it needs to understand the communication 
+protocol and the authentication scheme, verify the authenticity of a source and deal with rate limits and retries.
+In the FluxCD organization there are currently two operators that perform such operations. Both Flux and 
+Helm Operator connect to Git repositories to fetch Kubernetes objects, they need to maintain an up-to-date mirror 
+of one or several repos. Besides Git, Helm Operator needs to connect to Helm repositories hosted on public or 
+private HTTPS servers.
+
+## Motivation
+
+Each Flux or Helm Operator instance maintains its own Git repository mirror even if all of them
+point to the same source. If the Git repository host becomes unavailable, the cluster state will diverge from the last
+know desired state since the operators will stop the reconciliation due to pull errors. 
+
+Decoupling the Kubernetes objects acquisition from the reconciliation process with an in-cluster 
+source manager would make Flux and Helm Operator resilient to outbound connectivity issues and would
+simplify the state machine(s) that these controllers operate.
+
+Managing the source operations in a dedicated controller could allow Flux to compose the desire state of a cluster
+from multiple source.
```

Comment:
I don't see how it _allows_ this -- it's allowed already, it would just need some work.

Reply:
Is `enable` a better word than `allows`?

Reply:
I think the idea you're getting at is that it breaking the source management out would make the job of rewriting Flux to do multiplexing easier. In which case, "enable" is in the right spirit.

Reply:
I've changed it. Thanks